pkl/evaluator_manager_exec.go

```diff
@@ -206,5 +206,26 @@ func (e *execEvaluator) deinit() error {
 	close(e.out)
 	close(e.closed)
 	// TODO: graceful shutdown
-	return e.cmd.Process.Kill()
+	if err := e.cmd.Process.Signal(os.Interrupt); err != nil {
+		internal.Debug("Failed to interrupt process: %v", err)
+		return e.cmd.Process.Kill()
+	}
+	select {
+	case <-time.After(5 * time.Second):
+		// If the process does not exit within the timeout, kill it.
```

Comment:
I think this warrants logging even if the kill succeeds
```suggestion
	case <-time.After(5 * time.Second):
	    internal.Debug("Process did not exit within the timeout; killing.")
		// If the process does not exit within the timeout, kill it.
```